src/Microsoft.AspNetCore.OData/Adapters/WebApiAssembliesResolver.cs

```diff
@@ -47,11 +48,22 @@ public IEnumerable<Assembly> Assemblies
                 if (this.innerManager != null)
                 {
                     IList<ApplicationPart> parts = this.innerManager.ApplicationParts;
-                    return parts.Where(p => p is AssemblyPart).Select(p => (p as AssemblyPart).Assembly).Distinct();
+                    IEnumerable<Assembly> assemblies = parts
+                        .Where(p => p is AssemblyPart)
+                        .Select(p => (p as AssemblyPart).Assembly)
+                        .Distinct();
```

Comment:
nit: use `OfType<AssemblyPart>` to save the need of doing a `p is AssemblyPart` check?

i.e.
```
IEnumerable<Assembly> assemblies = parts
    .OfType<AssemblyPart>()
    .Select(p => p.Assembly)
    .Distinct();
```